management/src/main/java/io/micronaut/management/endpoint/management/ManagementEndpoint.java

```diff
@@ -0,0 +1,35 @@
+package io.micronaut.management.endpoint.management;
+
+import io.micronaut.http.hateoas.Resource;
+import io.micronaut.management.endpoint.annotation.Endpoint;
+import io.micronaut.management.endpoint.annotation.Read;
+import io.micronaut.web.router.UriRoute;
+
+import java.util.stream.Stream;
+
+/**
+ * <p>Exposes an {@link Endpoint} to list the available management endpoints of the application</p>
+ *
+ * @author Herán Cervera
+ * @since 2.5
+ */
+@Endpoint(ManagementEndpoint.ENDPOINT_ID)
+public class ManagementEndpoint {
+
+    static final String ENDPOINT_ID = "management";
+
+    private final ManagementRoutesResolver managementRoutesResolver;
+    private final ManagementLinkCollector managementLinkCollector;
+
+    public ManagementEndpoint(ManagementRoutesResolver managementRoutesResolver,
+                              ManagementLinkCollector managementLinkCollector) {
```

Comment:
It shouldn't be assumed the data being returned is links. Users may want to replace the implementation with something that does something entirely different. You could rename this to `ManagementDataCollector` to be generic

Reply:
I've renamed `ManagementLinkCollector` to `ManagementDataCollector`